### PR TITLE
Cleanup/remove effect of memory hint to write buffer and adjust some db parameters

### PR DESCRIPTION
**Pull request by asdacap**
Partially addresses #4040 

- Remove the effect of memory hint to the write buffer size of the databases. Memory hint still effect size of rowcache and shared block cache. 
- This is because write buffer size is quite significant to the database compaction as it effect the size of l0 file.
- Also, @benaadams seems to have better performance with lower memory hint. It could be due to the write buffer instead of  the block cache. It could be the bloom filter of the write buffer. It could be the size of the write buffer. Its hard to know for sure when both of them are binded to the same memory hint number.
- Additionally, expose some more config which can have an effect when upgrading to new rocksdb.
- Also expose the `AdviseRandomOnOpen` option. It looks like this does not have an effect on windows. So disabling it on linux and halfpath might improve performance which explain why @benaadams have much better performance. This definitely will cause hashdb to slowdown.
- Retuned BlockNumbers and CodeDb for low block size and have some row cache.
- StateDb's level multiplier and base size not changed as increasing it to have 3 level causes slightly slower sync. Will change in halfpath/flatdb as they naturally have lower write amplification rate.

#### What types of changes does your code introduce?

- [X] Refactoring

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Very slightly reduced time during code lookup.

### CODE REVIEW

**Comment by benaadams** [comment]
> Also, @benaadams seems to have better performance with lower memory hint. 

Was better during block processing generally, but terrible during OldReceipts blockprocessing

**Comment by asdacap** [comment]
Reducing block cache size definitely improve block processing time. IOPs measured from process increased, but IOPs measured from SSD decreased, meaning it make better use of OS cache. Using direct IO have the opposite effect as direct IO bypass os cache.

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 1]
```diff
@@ -378,15 +378,31 @@ private long GetMemtableSize()
         }
 
         tableOptions.SetFormatVersion(5);
-        tableOptions.SetFilterPolicy(BloomFilterPolicy.Create(10, false));
+        if (dbConfig.BloomFilterBitsPerKey.GetValueOrDefault() != 0)
+        {
+            if (dbConfig.UseRibbonFilterStartingFromLevel != null)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:383]: is not null

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs]: Done

**Pull request merged**